### ISSUE TITLE
Active class for links with query params

### ISSUE THREAD

**Issue by Rhincodon** [vuejs/vue-router#209]
HI. I have v-link to named route without query params. Active class works correctly on exact matching and within subroutes. But if I go to the same route with query params, active class no longer works.

Example:
I have link to users list in sidebar — `/users`. Active class works when route match `/users` and on subroutes like `/users/1`. But it does not work within route like `/users?page=2`.

On version 0.6.1 that behavior worked correctly, but on 0.7.5 it no longer works.

**Comment by Rhincodon** [comment]
Now active class for routes like `/users?page=2` working. But active class for links to inner routes like `/users/1` or `/users/1?page=1` no longer works if we have `params` AND `query` in named `v-link`. If we have just `params` it works. In previous 0.7.5 both `params` and `query` works together well, and active class toggled as needed.

**Comment by yyx990803** [comment]
I don't quite understand your description. Can you rephrase it with more details?

**Comment by Rhincodon** [comment]
Still not working previous comment. I created fiddle to illustrate what I mean `https://jsfiddle.net/co5mmz46/`. When we on `bar` route, `bar` link is not active. If we remove `query` on line 5 it will work. But if we have both `params` and `query` in v-link, active class not working.

**Comment by pajter** [comment]
It's still not working for me. Shouldn't the added `let dest` also be used in the else clause on line 139? https://github.com/vuejs/vue-router/blob/5ade5804bdbd7b71415a4cef4b3d227be0547fd4/src/directives/link.js#L139

**Comment by pajter** [comment]
This is still not working for me. I'm seeing in my debug console that the regex is testing for '^url' while `path` contains '/url'. So that regex because of the leading slash...

**Comment by yyx990803** [comment]
@pajter please open a separate issue with a reproduction using the latest version.
